src/license_guc.c

```diff
@@ -15,90 +15,84 @@
-		return true;
-
-	return (ts_guc_license_key == NULL || TS_CURRENT_LICENSE_IS_APACHE_ONLY()) &&
-		   tsl_handle == NULL;
-}
-
-/*
  * GUC checks work in two parts:
+ *
  *   1. In the check function, all validation of the new value is performed
- *      and any auxiliary state is setup but _not_ installed. Postgres is
- *      allowed to call the check function in cases where the newval will never
- *      be used, so any effects of this function must be confined to `extra` or
- *      otherwise be un-observable. Further, this function is _not_ allowed to
- *      throw exceptions.
+ *      and any auxiliary state is setup but not installed. This function
+ *      is not allowed to throw exceptions.
+ *
  *   2. In the assign function all user-visible state is installed. This
  *      function *MUST NOT FAIL* as it can be called from such places as
  *      transaction commitment, and will cause database restarts if it fails.
  *
- * Therefore our license validation also works in two parts, corresponding to
+ * Therefore license validation also works in two parts, corresponding to
  * check and assign:
- *   1. In the first stage we check the license kind, load the submodule into
- *      memory if needed (but don't link any of the cross-module functions yet)
- *      and, if it's an enterprise license, validate the license key.
- *   2. In the second stage we link all of the cross-module functions.
  *
- * The first stage will fail if the provided license key is invalid, or it's
- * trying to downgrade from an edition that uses the submodule
- * (Community and Enterprise) to one that does not (ApacheOnly). Currently only
- * upgrading is allowed within a session; downgrading requires starting a new
- * session.
+ *   1. In the first stage we check the license type, load the submodule into
+ *      memory if needed (but don't link any of the cross-module functions yet).
+ *
+ *   2. In the second stage we link all of the cross-module functions and init
+ *      tsl module.
  *
  * In order for restoring libraries to work (e.g. in parallel workers), loading
  * the submodule must happen strictly after the main timescaledb module is
  * loaded. In order to ensure that the initial value doesn't break this, we
  * disable loading submodules until the post_load_init.
+ *
+ * No license change from user session is allowed. License can be changed only
+ * if it is set from server configuration file or the server command line.
  */
 
+typedef enum
+{
+	LICENSE_UNDEF,
+	LICENSE_APACHE,
+	LICENSE_TIMESCALE
+} LicenseType;
+
+static LicenseType
+license_type_of(const char *string)
+{
+	if (string == NULL)
+		return LICENSE_UNDEF;
+	if (strcmp(string, TS_LICENSE_TIMESCALE) == 0)
+		return LICENSE_TIMESCALE;
+	if (strcmp(string, TS_LICENSE_APACHE) == 0)
+		return LICENSE_APACHE;
+	return LICENSE_UNDEF;
+}
+
+bool
+ts_license_is_apache(void)
+{
+	return license_type_of(ts_guc_license) == LICENSE_APACHE;
+}
+
 TSDLLEXPORT void
```

Comment:
I suggest to remove `TSDLLEXPORT` for consistency with the rest of the code. Sorry for the late comment about this, I was waiting for the response to Erik's comment.
```suggestion
void
```

Reply:
This function is used in TSL module, so keeping this export macro. I removed one from the function above